velox/exec/tests/ContainerRowSerdeTest.cpp

```diff
@@ -144,5 +144,36 @@ TEST_F(ContainerRowSerdeTest, nested) {
   testRoundTrip(nestedArray);
 }
 
+TEST_F(ContainerRowSerdeTest, compareDoubleEquals) {
+  static constexpr CompareFlags flags{true, true, false, false};
+  auto data = makeFlatVector<double>(
+      {std::numeric_limits<double>::quiet_NaN(), 2.0, 3.0});
+
+  // Test byte stream and array input.
+  ByteStream left;
+  auto leftPosition = serialize(data);
+  HashStringAllocator::prepareRead(leftPosition.header, left);
+
+  SelectivityVector allRows(data->size());
+  DecodedVector decodedVector(*data, allRows);
+  for (auto i = 0; i < data->size(); i++) {
+    EXPECT_EQ(0, ContainerRowSerde::compare(left, decodedVector, i, flags));
+  }
+
+  // Test byte stream inputs.
+  // Reset the byte stream.
+  leftPosition = serialize(data);
+  HashStringAllocator::prepareRead(leftPosition.header, left);
+
+  auto rightPosition = serialize(data);
+  ByteStream right;
+  HashStringAllocator::prepareRead(rightPosition.header, right);
+  auto type = data->type();
+
+  for (auto i = 0; i < data->size(); i++) {
```

Comment:
Try different combinations where Nan is compared with non-Nan values also as those code paths have changed now.

Also add tests for ascending/descending variation of flags.